### PR TITLE
Add null checks for entity on log archiver.

### PR DESCRIPTION
**Pull request by froboy**
**Related Issue/Ticket:**

Resolves https://github.com/ymcatwincities/openy_gated_content/issues/96

## Steps to test:

- [ ] Enable the VY log module
- [ ] Create some recurring Live Streams/Virtual Meetings
- [ ] View those nodes through VY
- [ ] Change the recurrence so that those nodes are deleted from the db
- [ ] Run a log export and see that it completes successfully.

## Quality checks:

Please check these boxes to confirm this PR covers the following cases:

- Maintaining our upgrade path is essential. Check one or the other:
  - [ ] This PR  provides updates via `hook_update_N` or other means.
  - [x] No updates are necessary for this change.
- Front end fixes should be tested against all of the Open Y Themes.
  - [ ] Tested against Carnation
  - [ ] Tested against Lily
  - [ ] Tested against Rose
  - [x] This change does not contain front-end fixes.
- [x] I have flagged this PR "Needs Review" or pinged the VY devs/QA
  team in Slack

### CODE REVIEW

**Review by hamrant — APPROVED** [review]
Code looks ok

**Comment by hamrant** [comment]
@froboy tested on build works fine.

I think it better to use an early return pattern, because above we already have predefined row values:

```php
      $export_row = [
        'created' => date('m/d/Y - H:i:s', $log->get('created')->value),
        'user' => $log->get('uid')->target_id ? $log->get('uid')->entity->getEmail() : '',
        'event_type' => $event_type,
        'entity_type' => $entity_type,
        'entity_bundle' => $entity_bundle,
        'entity_id' => $entity_id,
        'entity_title' => '',
        'entity_instructor_name' => '',
        'entity_created' => '',
      ];

...

        $entity = $this->entityTypeManager->getStorage($entity_type_id)
          ->load($entity_id);

        if (!$entity) {
          continue;
        }

```

**Comment by froboy** [comment]
@hamrant that's quite elegant. I haven't used `continue` very much, so I appreciate the suggestion. Updated.

File: modules/openy_gc_log/src/LogArchiver.php [hunk 1]
```diff
@@ -284,6 +284,9 @@ protected function prepareLogsForExport($logFileName = '') {
         $entity_type_id = $entity_type === 'node' ? 'node' : 'eventinstance';
         $entity = $this->entityTypeManager->getStorage($entity_type_id)
           ->load($entity_id);
+        if ($entity) {
```
Comment (anpolimus) [inline modules/openy_gc_log/src/LogArchiver.php:287]: @froboy , @hamrant we need something like this in this place:
```suggestion
        if ($entity is not instanceof EntityInterface) {
```

**Review by anpolimus — CHANGES_REQUESTED** [review]

**Comment by hamrant** [comment]
retest

**Comment by hamrant** [comment]
retest

File: modules/openy_gc_log/src/LogArchiver.php (hunk 1 above)
Comment (anpolimus) [reply modules/openy_gc_log/src/LogArchiver.php]: @hamrant I've updated it according to php standard
previous was producing fatal error:)

**Comment by hamrant** [comment]
@froboy please fix to

```php
use Drupal\Core\Entity\EntityInterface;

...

        if (!$entity instanceof EntityInterface) {
          continue;
        }
```

And make force push to remove the commit from @anpolimus

**Review by hamrant — DISMISSED** [review]

**Review by podarok — APPROVED** [review]

**Review by podarok — DISMISSED** [review]

**Review by hamrant — APPROVED** [review]

**Comment by NadyaRudenko** [comment]
Tested, no error occurs while I export logs. 
But in the exported logs the viewed live stream that I changed is not present. I don't know if this is expected behavior, so want a confirmation. Thanks!
image
image

**Comment by anpolimus** [comment]
retest

**Comment by NadyaRudenko** [comment]
retest

**Pull request merged**